solidity/contracts/transformers/wstETHTransformer.sol

```diff
@@ -105,6 +114,27 @@ contract wstETHTransformer is BaseTransformer {
     _neededUnderlying = Math.mulDiv(_expectedDependent, _totalSuppy, _totalShares, Math.Rounding.Up);
   }
 
+  function _takewstETHFromSenderAndUnwrap(
+    address _dependent,
+    uint256 _amount,
+    address _recipient
+  ) internal returns (uint256 _underlyingAmount) {
+    IwstETH(_dependent).safeTransferFrom(msg.sender, address(this), _amount);
+    _underlyingAmount = IwstETH(_dependent).unwrap(_amount);
```

Comment:
Would have been cool that wstETH had some kind of way of saving this gas.

Reply:
Yeah, definitely. But sadly, it doesnt